HeterogeneousCore/SonicTriton/interface/grpc_client_gpu.h

```diff
@@ -0,0 +1,7 @@
+#ifndef HeterogeneousCore_SonicTriton_grpc_client_gpu
+#define HeterogeneousCore_SonicTriton_grpc_client_gpu
+
+#define TRITON_ENABLE_GPU
+#include "grpc_client.h"
```

Comment:
You're right, that would potentially give weird errors (because the shared library is compiled with the actual CUDA symbols). I think the only way to do it fully consistently would be to modify the installed header file when building the external, so I'll update https://github.com/cms-sw/cmsdist/pull/6929/files accordingly.